### PR TITLE
Login UI and Global State

### LINKED ISSUE #3: Global Gamestate
A method for a global Gamestate should be determined and implemented.

The best solution for this may have changed with the introduction of the ECS system, so please research before implementing.

### LINKED ISSUE #4: Login UI and Function
- A Login UI should be created
- The Login UI should be hooked into account service requests.
- Error handling should be considered with the UI implementation.

### PR DESCRIPTION
**Pull request by ChristopherJMiller**
- Added ECS UI System for Forms
- Added Login UI
- Added Login Processing
- Added Global State for storing token when successfully logging in
- Added Tests for UI, Global State, and Login Processing

Closes #3 #4

### CODE REVIEW

File: Assets/Scripts/Global/ApplicationStateCreationSystem.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+using Unity.Entities;
+using Unity.NetCode;
+
+[UpdateInWorld(UpdateInWorld.TargetWorld.Default)]
+public class ApplicationStateCreationSystem : SystemBase {
+  internal struct CreateApplicationStateSingleton : IComponentData { }
```
Comment (aliceagain) [inline Assets/Scripts/Global/ApplicationStateCreationSystem.cs]: Nit: Rename to something that corresponds more to this Component's intended function.

File: Assets/Scripts/WWW/WebRequestComponents.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+using Unity.Collections;
+using Unity.Entities;
+
+public enum WebVerb {
```
Comment (aliceagain) [inline Assets/Scripts/WWW/WebRequestComponents.cs]: Nit: I thought these were called actions?

File: Assets/Tests/Utils/ECSTestBase.cs [hunk 3]
```diff
@@ -13,6 +13,7 @@ public abstract class ECSTestBase {
   [SetUp]
   public void SetUpBase() {
     w = new World("Default");
+    World.DefaultGameObjectInjectionWorld = w;
     em = w.EntityManager;
```
Comment (aliceagain) [inline Assets/Tests/Utils/ECSTestBase.cs]: I think it would be better to name these something understandable like testWorld or just world.

**Review by aliceagain — DISMISSED** [review]
Just a few minor changes.

File: Assets/Tests/Utils/ECSTestBase.cs (hunk 3 above)
Comment (ChristopherJMiller) [reply Assets/Tests/Utils/ECSTestBase.cs]: fixed in 6c9e816

File: Assets/Scripts/WWW/WebRequestComponents.cs (hunk 2 above)
Comment (ChristopherJMiller) [reply Assets/Scripts/WWW/WebRequestComponents.cs]: fixed in 6c9e816

File: Assets/Scripts/Global/ApplicationStateCreationSystem.cs (hunk 1 above)
Comment (ChristopherJMiller) [reply Assets/Scripts/Global/ApplicationStateCreationSystem.cs]: fixed in 6c9e816

**Pull request merged**